grid/src/js/views/form.js

```diff
@@ -0,0 +1,308 @@
+              <label class="invalid-message" for="price"></label>
+            </div>
+          </div>
+          <div class="form-group form-row">
+            <div class="col-7">
+              <select class="form-control" id="delivery" name="delivery">
+                <option value="null" ${!hasDelivery ? 'selected' : ''}>None delivery</option>
+                <option value="country" ${hasDelivery ? 'selected' : ''}>Country</option>
+                <option value="city">City</option>
+              </select>
+            </div>
+          </div>
+          <div class="form-group form-row">
+            <div class="col-7">
+              <fieldset class="form-fieldset" id="country" ${!hasDelivery ? 'style = "display: none"' : ''}>
+                <legend class="form-legend">Country</legend>
+                <div id="country-list">${deliveryTemplate.countryList}</div>
+              </fieldset>
+              
+              <fieldset class="form-fieldset" id="city" style = "display: none">
+                <legend class="form-legend">City</legend>
+                <div id="city-list">${deliveryTemplate.cityList}</div>
+              </fieldset>
+            </div>
+            
+            <div class="col align-self-center">
+              <label class="invalid-message" for="city"></label>
+            </div>
+            
+          </div>
+          <div class="form-group mb-0">
+            <button class="btn btn-success" type="button" id="saveProduct" data-product="${id ? id : ''}">
+              <span class="spinner-grow spinner-grow-sm d-none" role="status" aria-hidden="true"></span>
+              Save
+            </button>
+          </div>
+        </fieldset>
+      </form>
+    </div>`;
+
+    return template;
+  }
+
+  setInvalidFocus() {
+    let self = this;
+    self.$formChange.find('.invalid-input').first().focus();
+  }
+
+  toggleValidationError(input, error = '') {
+    let self = this;
+
+    self.$formChange = $('#formChange');
+
+    // debugger;
+    let $error = self.$formChange.find(`label[for=${input}]`);
+    let $input = self.$formChange.find(`#${input}`);
+
+    $error.html(error);
+
+    if (error && !$input.hasClass('invalid-input')) $input.addClass('invalid-input');
+    else if (!error && $input.hasClass('invalid-input')) $input.removeClass('invalid-input');
+  }
+
+  toggleBtnDisable($btn, productId = null) {
+    let isDisable = $btn.prop('disabled');
+
+    $btn.closest('fieldset').prop('disabled', !isDisable);
+    $btn.prop('disabled', !isDisable);
+    $btn.children('.spinner-grow').toggleClass('d-none');
+
+    if (productId) {
+      $(`#tableProducts #${productId}`).find('button').prop('disabled', true);
+    }
+  }
+
+  triggerSaveProduct() {
+    let self = this;
+
+    if (!self.$saveBtn) self.init();
+    let data = self.$formChange.serializeArray();
+
+    let product = self.$saveBtn.attr('data-product') ? self.$saveBtn.attr('data-product') : null;
```

Comment:
There's dedicated function for data attributes. 
https://api.jquery.com/jQuery.data/